shared/src/ledger/eth_bridge/bridge_pool_vp.rs

```diff
@@ -115,43 +115,33 @@ where
             }
         };
 
-        // check that only the pending_key value is changed
-        if keys_changed.iter().any(is_protected_storage) {
-            tracing::debug!(
-                "Rejecting transaction as it is attempting to change the \
-                 bridge pool storage other than the pending transaction pool"
-            );
-            return Ok(false);
+        let pending_key = get_pending_key(&transfer);
+        for key in keys_changed.iter().filter(|k| is_bridge_pool_key(k)) {
+            if *key != pending_key {
+                tracing::debug!(
+                    "Rejecting transaction as it is attempting to change an \
+                     incorrect key in the pending transaction pool: {}.\n \
+                     Expected key: {}",
+                    key,
+                    pending_key
+                );
+                return Ok(false);
+            }
```

Comment:
Should we reject if there is already a value at `read_pre_value(&pending_key)` to prevent the same burn being added twice?

Reply:
yes we should